clients/drcachesim/drpt2trace/pt2ir.h

```diff
@@ -0,0 +1,316 @@
+/* **********************************************************
+ * Copyright (c) 2022 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* pt2ir: convert a PT raw trace to DynamoRIO's IR format. */
+
+#ifndef _PT2IR_H_
+#define _PT2IR_H_ 1
+
+/**
+ * @file drmemtrace/pt2ir.h
+ * @brief Offline PT raw trace converter.
+ */
+
+#include <string>
+#include <vector>
+#include <memory>
+#define DR_FAST_IR 1
+#include "dr_api.h"
+
+#ifndef IN
+#    define IN // nothing
+#endif
+#ifndef OUT
+#    define OUT // nothing
+#endif
+#ifndef INOUT
+#    define INOUT // nothing
+#endif
+
+/**
+ * The type of pt2ir_t::convert() return value.
+ */
+enum pt2ir_convert_status_t {
+    PT2IR_CONV_SUCCESS = 0,       /**< The conversion process is successful. */
+    PT2IR_CONV_ERROR_SYNC_PACKET, /**< The conversion process ends with a failure to sync
```

Comment:
Since the comments mostly don't fit in one half-line I would suggest moving them all to pre-field whole-line comments to avoid these multi-partial-line situations which seem awkard to me.

Reply:
Done.